pkg/apphealth/status.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package apphealth
+
+import "time"
+
+const (
+	// reportMinInterval is the minimum interval between health reports.
+	reportMinInterval = time.Second
```

Comment:
q: why are we reporting the health at second interval? Seems wasteful to be waking this go routine when there is no change. Why do we not report only where there is a change?

Reply:
this is existing code that I just moved to a more logic file. I did not write this nor change it to minimize changes that my PR introduced. This is a great follow up to be done in a separate issue/pr

Reply:
Can we change this now since we are already changing the data structure?

Reply:
done